### ISSUE TITLE
Explore how to solve the problems in forming and changing partnerships between schools, LPs and DPs

### ISSUE THREAD

**Issue by claire-hughez** [DFE-Digital/ecf2#35]
## Why

Schools are having lots of problems in ECF1 changing their partnerships with lead providers and delivery partners.

Currently, once a partnership is outside rejection window:

* schools cannot currently remove or edit their lead provider and delivery partner in the partnership
* providers can only edit the delivery partner (via the API)
* DfE admin users have to remove a partnership in the admin UI to then allow a new lead provider to report a partnership over the API

However, this has meant at the start of a new academic year, many schools have struggled to get the changes made to their lead provider and delivery partner. Lead providers have not made the changes yet or asked DfE admin users to ‘release’ the schools so they can work with other lead provider and delivery partner combos.

## What

We want to consider a solution where schools could partner with a lead provider and delivery partner, rather than appointing them outside of the service and waiting for confirmation in the service.

Lead providers could still give this information as well, but we think schools should be able to change it too when necessary. We should explore any problems in having 2 potential owners of this data.

Post-MVP this might require more complicated functionality for the partnerships created by schools (e.g. a “proposed”/”tentative” status), but for MVP these school-created partnerships would look, smell and taste just like a provider-confirmed partnership.

If we were to consider a solution where schools stated the LP and DP, we also need to consider how lead providers could challenge this over the API if it was not valid.

## Context & links

None yet - can try and find links to the problems this year. (Colin/Bernard/Anna). [NE - It was also uncovered as a pain point in the 2023 discovery]

## Who knows about this

@easeynathan

Semi-Final lucid
https://lucid.app/lucidspark/94ab3793-b960-403b-b598-3676caf8ccb3/edit?view_items=pfc-Hk2izg8~&invitationId=inv_87d98967-2b2c-4df5-8cd7-620e14fc8093

**Comment by peteryates** [comment]
> We want to consider a solution where schools could partner with a lead provider and delivery partner, rather than appointing them outside of the service and waiting for confirmation in the service.

I like the idea of this and the [draft data model](https://github.com/DFE-Digital/ecf2/discussions/2#discussioncomment-10008460) as it stands would support it nicely.

In the most recent version we added many-to-many join tables between `School settings` and `Provider partnership`. This could allow a school to have several partnerships to choose from. 

![image](https://github.com/user-attachments/assets/2195d295-13f8-4409-9a4b-9d1daeafa472)

It could also allow for a proposal/confirmation type interaction like you describe above - this could happen in both directions if necessary (provider proposes and school confirms or vice versa).

**Comment by mason-emily** [comment]
No. schools that have challenged a partnership (per cohort):

2021 - 949 schools; 2022 - 757 schools; 2023 - 449 schools

**Comment by claire-hughez** [comment]
https://educationgovuk.sharepoint.com/:w:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Schools%20and%20Teachers%20Team/Analysis/Partnership%20Reporting/Supporting%20schools%20in%20reporting%20partnerships.docx?d=w29620c0e4b345a2d4c6fc48ff63f29f9&csf=1&web=1&e=hJ1GR1

**Comment by kas-madrigal** [comment]
happy to pick this up on next sprint

**Comment by kas-madrigal** [comment]
Reminder - 

- do we need to restructure any database user hierarchy rules if ABs are also often DPs but they can't provide for their own school? @peteryates 
-  could we remind ourselves of the rules for choosing DPs and ABs for schools (if they use AB 1 for ITT they can't use the same one for induction) @mason-emily we can look through service rules

**Comment by kas-madrigal** [comment]
Relevant reading

- https://lucid.app/lucidspark/61119b71-e7f8-4337-9809-a82362977ac7/edit?view_items=0Fk8.D7cHJu~&invitationId=inv_3aaebc47-c7a4-44ff-8600-0aff4802d863 

- https://lucid.app/lucidspark/61119b71-e7f8-4337-9809-a82362977ac7/edit?view_items=0Fk8.D7cHJu~&invitationId=inv_3aaebc47-c7a4-44ff-8600-0aff4802d863

- https://educationgovuk.sharepoint.com/:w:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/CPD%20Technical%20Architecture/04.%20Assurance/Data%20privacy%20impact%20assessment/history/Contract%20Responsibilities%20Analysis.docx?d=waa273ed47cd648519a9868825716ede1&csf=1&web=1&e=S3aGW7

- https://educationgovuk.sharepoint.com/:w:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Schools%20and%20Teachers%20Team/Analysis/Lead%20Provider_Delivery%20Partner%20Changes.docx?d=w872b25810b575c669e00a9add409ebab&csf=1&web=1&e=Kpriqb

- https://educationgovuk.sharepoint.com/:w:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/10.%20CPD%20Lead%20Provider/Delivery/02_Delivery_Lead%20Provider%20team%20induction%20guide.docx?d=wb08b4f8c36135915bc901307fa66b690&csf=1&web=1&e=d2398Z

**Comment by kas-madrigal** [comment]
Collecting learnings here https://lucid.app/lucidspark/94ab3793-b960-403b-b598-3676caf8ccb3/edit?view_items=Qji9_EB3EneQ&invitationId=inv_87d98967-2b2c-4df5-8cd7-620e14fc8093

**Comment by gracecbrennan** [comment]
This is ongoing

**Comment by claire-hughez** [comment]
Some bits Nathan flagged with me that are worth looking into:
- New partnerships will need to be backdated to previous contracts potentially, as schools might move their LP/DP for ECTs part way through
- Academic year obviously impacts the partnerships available for schools to choose, and might get confusing, so good to think through that... Although maybe less necessary based on the above? One to pick up and analyse maybe!

**Comment by gracecbrennan** [comment]
Aiming to complete today, with support from Kerry and Nathan

**Comment by kas-madrigal** [comment]
I've turned the summary of work into a PDF because it's the middle ground between my visual bran and other team member's more words-based brains. [The original lucid is here. ](https://lucid.app/lucidspark/94ab3793-b960-403b-b598-3676caf8ccb3/edit?view_items=pfc-Hk2izg8~&invitationId=inv_87d98967-2b2c-4df5-8cd7-620e14fc8093) 

I've learned 3 very important things throughout this process.

1. Partnership reporting will work well ONLY if other steps of the process work well. This is because we depend on ABs on the AB portal to record cohort and academic year, which are intrinsic to correct partnership reporting. LP use this data to then submit a partnership and connect those data points to contracts but the initial inputs happen earlier on in the process. 
2. As with induction data we have an overly-complicated system of time tracking that requires input from multiple sources isolated from each other (schools, ABs and LP). They are in different systems, happening in a non-sequential way for a system that requires them to be very sequential in order to succeed. On top of this, users don't understand the entire process and do not understand our requests or their responsibility in them. 
3. Redesign of induction data and LP partnership should come hand in hand as LP partnership is last the service journey flow and uses the same data points. If induction data and time tracking is redesign correctly, partnership user flow could be much easier for us and for users. 


[Partnership research ECF 2 - Page 1 Frame 7.pdf](https://github.com/user-attachments/files/16898979/Partnership.research.ECF.2.-.Page.1.Frame.7.pdf)
